src/client/package.json

```diff
@@ -0,0 +1,3 @@
+{
+  "type": "module"
```

Comment:
This is required to make sure it will always work to make `.js` considered as ESM, whatever the user's package type.

Reply:
We actually don't need this here, this is auto generated by our vite plugin while building.

Reply:
<img width="273" alt="image" src="https://user-images.githubusercontent.com/8336744/183278360-d5bca4a9-4ff1-44a9-8fe4-ce90aee484e1.png">

I didn't see it when I remove this and run `pnpm run docs`, is there anything I missed?

Reply:
I meant when you build the site, it will be generated in the .temp directory. Why do we need this in dist/client?

Reply:
> This is required to make sure it will always work to make .js considered as ESM in temporary client directory, no matter whatever the user's package type.

Reply:
Yeah I read that 😅, but its still not clear. Any particular example where not having this will break stuff?

Reply:
When the user is not using `type: module` or using `type: commonjs`, all `.js` will be treated as commonjs by default, unless there is a closer `package.json` declaring `type: module` all `.js` should be considered as ESM.

```plan
- docs
  - .vitepress
    - .temp
      pacage.json # `type: "module"` is required
- package.json # `commonjs` will also affects `docs/.vitepress/.temp` (I forgot the real path here, the depth doesn't matter)
```

Reply:
Yeah inside that .temp we are emitting this file already: https://github.com/vuejs/vitepress/blob/chore/vite-3/src/node/plugin.ts#L227

Reply:
> Yeah inside that .temp we are emitting this file already: [https://github.com/vuejs/vitepress/blob/chore/vite-3/src/node/plugin.ts#L227](https://github.com/vuejs/vitepress/blob/chore/vite-3/src/node/plugin.ts?rgh-link-date=2022-08-07T06%3A44%3A20Z#L227)

Thanks for pointing it out, I'm not so familiar with the code base. 🤣

Reply:
It will emit it by default, unless user has set ssr format in vite.config to cjs for some reasons. In that case we'll respect that.

Reply:
@brc-dd 

OK, let's see if it will cause some issues later. As I imagine, all `.js` in `.temp` are ESM now, so setting is as cjs is meaningless and won't work.

Reply:
All .js in temp are actually generated by vite, so if user has specified cjs there, then they would be cjs.

Reply:
Thanks for pointing!